### ISSUE TITLE
[CT-3186] [Bug] Compilation Error on macros during partial parsing

### ISSUE THREAD

**Issue by sanromeo** [dbt-labs/dbt-core#8775]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

While `run`, `compile` or other command for partial parsing received `Compilation Error`:
```
dbt found two macros named "<user_macro>" in the project "<user_project>"

To fix this error, rename or remove one of the following macros:
- /macros/foo/bar.sql
- /macros/foo/bar.sql
```

### Expected Behavior

No error

### Steps To Reproduce

1. Create macros.sql file with 2 macros inside
2. Create macros.yml with documenation for this 2 macros
3. Run `dbt compile`
4. Change something in macros.yml
5. Run `dbt compile` again

### Relevant log output

```shell
08:46:39  Encountered an error:
Compilation Error
  dbt found two macros named "<macros_name>" in the project "<project_name>".
   To fix this error, rename or remove one of the following macros:
      - macros/foo/bar.sql
      - macros/foo/bar.sql
08:46:39  Command `dbt compile` failed at 11:46:39.886922 after 1.85 seconds
08:46:39  Flushing usage events
```

### Environment

```markdown
- Python: 3.10.4
- dbt: 1.6.2
```

### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

dbt-adapter: dbt-athena: 1.6.1

**Comment by sanromeo** [comment]
Similar Isuue: #4233 
Related PR for #4233 which should have fixed #4773 but I guess it fixed only for snapshots not for macros.

**Comment by dbeatty10** [comment]
Thanks for reaching out about this @sanromeo! 🏆 

Re-opening https://github.com/dbt-labs/dbt-core/issues/4233 and closing this as a duplicate.
